src/notation/internal/notationactioncontroller.h

```diff
@@ -186,6 +190,9 @@ class NotationActionController : public actions::Actionable, public async::Async
     bool canUndo() const;
     bool canRedo() const;
     bool isNotationPage() const;
+    bool isStandardStaff() const;
+    bool isTablatureStaff() const;
+    void registerAction(const mu::actions::ActionCode&, std::function<void()>, bool (NotationActionController::*)() const);
```

Comment:
Maybe it makes sense, instead of `bool (NotationActionController::*)() const`, to make a more general solution, like `std::function<bool()>`

Reply:
They're both there...but I'd suggest you review my other PR first, it uses this idea but expands it considerably and simplifies the UiContext stuff too (no more "text editing" context for a start!).

Reply:
I looked, good solution!